src/intersection/compute_intersection.cc

```diff
@@ -301,24 +303,25 @@ IntersectionResult ComputeIntersection<Simplex<1>, Simplex<2>>::compute(std::vec
                     signed_plucker_product(1),
                     signed_plucker_product(2)};
     double w_sum = w[0] + w[1] + w[2];
-
+    
     unsigned int n_positive = 0;
     unsigned int n_negative = 0;
     unsigned int zero_idx_sum =0;
-    DebugOut().VarFmt(std::fabs(w_sum));
-    DebugOut().VarFmt(scale_line_);
-    DebugOut().VarFmt(scale_triangle_);
-    if(std::fabs(w_sum) > rounding_epsilon*scale_line_*scale_triangle_*scale_triangle_) {
+//     DebugOut().VarFmt(std::fabs(w_sum));
+//     DebugOut().VarFmt(scale_line_);
+//     DebugOut().VarFmt(scale_triangle_);
+    if(std::fabs(w_sum) > rounding_epsilon*scale_line_*scale_triangle_*scale_triangle_)
         w = w / w_sum;
-
-        for (unsigned int i=0; i < 3; i++) {
-            //DebugOut().VarFmt(i).VarFmt(w[i]);
-            if (w[i] > rounding_epsilon) n_positive++;
-            else if ( w[i] > -rounding_epsilon) zero_idx_sum+=i;
-            else n_negative++;
-        }
+        
+    for (unsigned int i=0; i < 3; i++) {
```

Comment:
This FOR cycle used to be under the IF condition before. I believe it was wrong. 
I had to solve problem, where products where -0.5, -0.5, 1.0 which summed to zero and let to degenerate case (and should not).
In compute_final and compute_degenerate I consider that the degenerate case means all three zero products.
Do you agree?
I add 12 unit test mesh for this.